### ISSUE TITLE
Improve magnetic detection for multichannel images

### ISSUE THREAD

**Issue by pearlryder** [piximi/image-viewer#79]
In a multi-channel microscopy image, it would be great to have access to individual channels to work on the edges of that channel (in the example image in Piximi, it'd be nice to run magnetic selection only on the blue DNA channel in order to segment the nuclei more easily, since these have a sharply defined edge in this channel only). 

![Screen Shot 2021-03-17 at 1 23 49 PM](https://user-images.githubusercontent.com/28375904/111525434-02cd7300-8734-11eb-80ca-2ea25fc0f0cd.png)

**Comment by DavidStirling** [comment]
We really should have a system to exclude "used" coordinates when drawing a second line. I expect this might be a graph cut kind of problem that @0x00b1 might know something about. Either we could "disconnect" the used coordinates from the graph or have some kind of distance penalty when backtracking close to an existing line.

**Comment by 0x00b1** [comment]
@DavidStirling I agree! But I think that is a separate issue from the issue @pearlryder is raising. I think the @pearlryder issue is related to sampling all channels rather than just one or two specific channels.

**Comment by 0x00b1** [comment]
I'm writing a sample list component that will allow users to choose what channels they want to sample from. Hopefully, this addresses the issue.

**Comment by DavidStirling** [comment]
I'm not entirely convinced that the individual channel sampling is what went wrong here. It looks more like the graph resolution is much lower than that of the image?

**Comment by DavidStirling** [comment]
That said, selectively sampling channels is still a very useful feature to add.

**Comment by pearlryder** [comment]
Yeah, I created a separate issue that relates to what you're discussing regarding the graph coordinates @DavidStirling - the picture isn't the best example wanting to work w/ individual channels, but Allen and Alice figured it out!
